### ISSUE TITLE
"Group not found or you don't have access" appearing inappropriately

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#10125]
Over the past couple of days, maybe longer, I've been seeing a lot of reports of players seeing "Group not found or you don't have access" when they've been trying to join public guilds - significantly more than I'd expect if it were just new players being confused about how to join guilds. These reports have been for the website and both mobile apps.

Those users often also report seeing the `userAlreadyInGroup` error message (indicating they're trying to join a group they're already a member of) but again I'm seeing this much more often than I'd expect from just users mistakenly trying to join guilds they're already in.

I'm pretty sure there's some bug somewhere but I haven't investigated further.

An example from User ID d1befce6-9299-487e-a687-e8ffa99f98f7: _"It's with trying to join a guild I get multiple errors I will attach a screenshot [screenshot showed `userAlreadyInGroup`]. The one error is that I'm already in the guild, which is confusing because I'm not. The other error looks very similar but says I do not have access into try again later"_

**Comment by Alys** [comment]
An example from User ID d1befce6-9299-487e-a687-e8ffa99f98f7: "It's with trying to join a guild I get multiple errors I will attach a screenshot [screenshot showed `userAlreadyInGroup`]. The one error is that I'm already in the guild, which is confusing because I'm not. The other error looks very similar but says I do not have access into try again later"

**Comment by Alys** [comment]
There was also this report in the Bug guild from Praxis Arete (0ce6d43b-efa2-45a3-acaa-59daad7d658d) about five days ago. It's about the userAlreadyInGroup problem more than the "not found / no access" error message but it might be related. The lag described is certainly odd.

_"I've noticed a latency issue between the time of joining a guild and when it stops displaying a "Join" button in "My Guilds" and also on the guild's home page. Yesterday this was just a few minutes, today it has been over half an hour since I joined -- and both locations still display a green "Join" button." The orange-red error message says ... "already in group" (plus some other text) in the oval when I click on the "Join" button again. ... this is on the website from a computer browser... The only times I've ever joined a guild were yesterday and the day before. After I had joined two guilds within a couple hours of each other, I saw that the more recent one still had a "Join" button in the "My Guilds" menu (a few minutes after joining). That gave me the error when I clicked on it. This happened again that day after I had joined a third guild (only for the most recent join) and then again last night right before I posted it here (since it seemed that the latency on removing the extra "Join" button was more than a few minutes). As far as I can tell, it is every time, although I did not try navigating via "My Guilds" until I had already joined a couple guilds. Once again, the extra "Join" button was always on the "My Guilds" page and at least a couple of times on the actual guild page when visited... but only for a while after joining. ... My OS is Win XP and browser is Seamonkey 2.48 (similar to Firefox). It's the latest version that supports Win XP. "_
